### PR TITLE
Calculate topology based on registered data nodes

### PR DESCRIPTION
**Pull request by lujiajing1126**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Fixes apache/skywalking#11238
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking-banyandb/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by lujiajing1126** [comment]
Generally be ready to be reviewed. Test cases still need to need polished.

File: test/integration/distributed/setup/node_test.go
Comment (lujiajing1126) [inline test/integration/distributed/setup/node_test.go:122]: Any idea how we can better organize such (fine-grained) services setup

Comment (hanahmily) [reply test/integration/distributed/setup/node_test.go:122]: Would you be able to use a mock to test the behavior of the selector? 

I have introduced a cluster integration in #333 . We can use it to test the cluster selector.

File: banyand/liaison/grpc/node.go
Comment (hanahmily) [inline banyand/liaison/grpc/node.go:50]: #333 will use `queue.Client` to listen to node events, reducing the need for the liaison node to set up two etcd watchers. Please merge this once PR is merged.

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Comment by lujiajing1126** [comment]
I would raise another PR since there are too many conflicts.

**Pull request closed without merging**
